server/routes/routes.js

```diff
@@ -82,52 +82,30 @@ router.get("/", (req, res) => {
 //   }
 // });
 
-router.post("/validate/local/multiple/:domain", async (req, res) => {
-  const { dirPath } = req.body;
-  const { domain } = req.params;
-  if (!domain || !dirPath)
-    return res
-      .status(400)
-      .json({ msg: 'Req Body needs to have "domain" and "dirPath"' });
+router.post("/validate/:domain", async (req, res) => {
+  try {
+    const { logPath } = req.body;
+    const { domain } = req.params;
 
-  if (!Object.keys(SUPPORTED_DOMAINS_SORTED_INDEX).includes(domain))
-    return res.status(404).json({ msg: `Domain ${domain} not supported yet!` });
+    /* -------- Input validation ------- */
+    if (!logPath)
+      return res.status(400).json({ msg: 'Req Body needs to have "logPath"' });
 
-  try {
-    const files = fs.readdirSync(dirPath);
-    const firstFileData = fs.readFileSync(
-      path.join(
-        dirPath,
-        files.filter((filename) => filename.match(/\.json$/))[0]
-      )
+    if (!Object.keys(SUPPORTED_DOMAINS_SORTED_INDEX).includes(domain))
+      return res
+        .status(404)
+        .json({ msg: `Domain ${domain} not supported yet!` });
+
+    /* ------ End of Input validation ------- */
+
+    const destination = await readLogFiles(domain, logPath);
+    await validateLogs(domain, logPath, destination);
```

Comment:
I discussed this with Abhik, however, can you please check if we can call index.js here, instead of directly calling validateLogs.

Reply:
Sorry, I didn't get you here

Reply:
As discussed check this once.

Reply:
I checked this and it is used to call the utility from the CLI only. So, i think we can't use it here.